### PR TITLE
Docker docs

### PR DESCRIPTION
**Pull request by vdantu**
*Issue #, if available:*
Docker docs .. First pass
*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docker/advanced_settings.md [hunk 1]
```diff
@@ -335,111 +295,43 @@ The predict endpoint will return a prediction response in JSON. It will look som
...
 ```
 
-
 ## Description of Config File Settings
 
-**For mms_app_cpu.conf:**
+**For mms_config.properties:**
 
-The system settings are stored in [mms_app_cpu.conf](mms_app_cpu.conf). You can modify these settings to use different models, or to apply other customized settings. The default settings were optimized for a c5.2xlarge instance.
+The system settings are stored in [mms_config.properties](mms_config.properties). You can modify these settings to use different models, or to apply other customized settings.
```
Comment (frankfliu) [inline docker/advanced_settings.md]: We can link to [mms_config.properties](../docker/config.properties)

File: docker/advanced_settings.md [hunk 2]
```diff
@@ -335,111 +295,43 @@ The predict endpoint will return a prediction response in JSON. It will look som
...
-```
-
-or
+in the range of 0 .. (num-gpu-1) in a round-robin fashion. **By default MMS uses all the available GPUs but this parameter can be configured if user want to use only few of them**. A discussion on how to set this parameter can be found [here](../docs/optimised_config.md)
 
-```bash
-curl -X GET http://your_public_host_name/ping
-```
+```yaml
+vmargs=-Xmx1g -XX:MaxDirectMemorySize=512m -Dlog4j.configuration=/opt/ml/conf/log4j.properties
```
Comment (frankfliu) [inline docker/advanced_settings.md]: Most of properties should be commented out.

File: docker/advanced_settings.md [hunk 3]
```diff
@@ -266,38 +229,35 @@ Similar to CPU base image the prebuilt GPU 'base' image is hosted at `awsdeeplea
...
 ### Testing the MMS Docker
 
-Now you can send a request to your server's [api-description endpoint](http://localhost/api-description) to see the list of MMS endpoints or [ping endpoint](http://localhost/ping) to check the health status of the MMS API. Remember to add the port if you used a custom one or the IP or DNS of your server if you configured it for that instead of localhost. Here are some handy test links for common configurations:
+Now you can send a request to your server's [api-description endpoint](http://localhost/api-description) to see the 
+list of MMS endpoints or [ping endpoint](http://localhost/ping) to check the health status of the MMS API. 
+Remember to add the port if you used a custom one or the IP or DNS of your server if you configured it for that instead 
+of localhost. Here are some handy test links for common configurations:
 
 * [http://localhost/api-description](http://localhost/api-description)
 * [http://localhost:8080/api-description](http://localhost/api-description)
```
Comment (frankfliu) [inline docker/advanced_settings.md]: Why keep 8080 port link here?

File: docker/README.md [hunk 4]
```diff
@@ -30,16 +30,17 @@ Running MXNet Model Server with Docker in two steps:
...
-With the `-p` flag, we're setting it up so you can run inference on your host computer's port: `80`.
+With the `-p` flag, we're setting it up so you can run inference on your host computer's port: `80` 
+and you can run other control plane API's on your host computer's port : `81`
 
 **Step 2: Test inference.**
 
 ```bash
 curl -O https://s3.amazonaws.com/model-server/inputs/kitten.jpg
-curl -X POST http://127.0.0.1/squeezenet/predict -F "data=@kitten.jpg"
+curl -X POST http://127.0.0.1/predictions/squeezenet -F "data=@kitten.jpg"
```
[the remark this answers is not in the document]
Comment (vdantu) [reply docker/README.md]: Fixed.

File: docker/README.md [hunk 5]
```diff
@@ -81,50 +81,31 @@ mkdir /tmp/models
...
-
 Download the template for a CPU or a GPU config and place it in the `models` folder you just created:
-* [mms_app_cpu.conf](mms_app_cpu.conf)
-* [mms_app_gpu.conf](mms_app_gpu.conf)
+* [mms_config.properties](mms_config.properties)
 
 **Step 3: Modify the configuration template.**
 
-Edit the file you downloaded, `mms_app_*pu.conf`. It will have the following section for `MMS Arguments`:
+Edit the file you downloaded, `mms_config.properties`. It will have the following section for `MMS Arguments`:
```
[the remark this answers is not in the document]
Comment (vdantu) [reply docker/README.md]: True. I think I missed it.

File: docker/README.md [hunk 6]
```diff
@@ -81,50 +81,31 @@ mkdir /tmp/models
 
 **Step 2: Download the configuration template.**
 
-To run a different model in the Docker image, you need to provide it a new configuration file.
-
 Download the template for a CPU or a GPU config and place it in the `models` folder you just created:
-* [mms_app_cpu.conf](mms_app_cpu.conf)
-* [mms_app_gpu.conf](mms_app_gpu.conf)
+* [mms_config.properties](mms_config.properties)
```
[the remark this answers is not in the document]
Comment (vdantu) [reply docker/README.md]: Yes, I had removed the files from reference, I missed the `CPU or a GPU config` from Readme. Fixed it.

File: docker/advanced_settings.md (hunk 1 above)
Comment (vdantu) [reply docker/advanced_settings.md]: Thats what the above does. It links to the local directory.

File: docker/README.md [hunk 7]
```diff
@@ -133,22 +114,26 @@ Save the file.
...
-docker run -itd --name mms -p 80:8080 -v /tmp/models/:/models awsdeeplearningteam/mms_cpu
+docker run -itd --name mms -p 80:8080 -p 81:8081 -v /tmp/models/:/models awsdeeplearningteam/mms_cpu
 ```
 
+**NOTE**: If you modify the inference_address or the management_address in the configuration file, accordingly modify the 
+ports exposed by docker as well.
+
 **Step 5: Start MMS.**
-You also need to start MMS in the container. In the Quickstart, we did this as all one command, but here you are doing it as a second step.
+You also need to start MMS in the container. In the Quickstart, we did this as all one command,
```
[the remark this answers is not in the document]
Comment (vdantu) [reply docker/README.md]: There is "bash -c " .. Do we need to add anything more?

File: docker/README.md [hunk 8]
```diff
@@ -30,16 +30,17 @@ Running MXNet Model Server with Docker in two steps:
 This will download the MMS Docker image and run its default configuration, serving a Squeezenet model.
 
 ```bash
-docker run -itd --name mms -p 80:8080 awsdeeplearningteam/mms_cpu mxnet-model-server start --mms-config /mxnet_model_server/mms_app_cpu.conf
+docker run -itd --name mms -p 80:8080 -p 81:8081 awsdeeplearningteam/mms_cpu mxnet-model-server --start --models squeezenet=https://s3.amazonaws.com/model-server/models/squeezenet_v1.1/squeezenet_v1.1.model
 ```
 
-With the `-p` flag, we're setting it up so you can run inference on your host computer's port: `80`.
+With the `-p` flag, we're setting it up so you can run inference on your host computer's port: `80` 
+and you can run other control plane API's on your host computer's port : `81`
```
[the remark this answers is not in the document]
Comment (vdantu) [reply docker/README.md]: This should be OK. I explained that this is mapping to some ports, so customers who want to run it on different ports would map to different ports.

File: docker/README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (frankfliu) [reply docker/README.md]: In earlier docker run command MMS already started with no model loaded,
So your 2nd command won't work.

My suggestion is merge them into single step.

File: docker/README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (frankfliu) [reply docker/README.md]: We should provide best practice even in sample command.

File: docker/advanced_settings.md (hunk 1 above)
Comment (frankfliu) [reply docker/advanced_settings.md]: I mean config.properties.

We should not rename it to mms_config.properties. This file is used in DockerFile and MMS will only pickup config.properties in working folder, not mms_config.properties.

File: docker/README.md [hunk 9]
```diff
@@ -81,50 +81,31 @@ mkdir /tmp/models
...
---log-file
-optional
-
---log-rotation-time
-optional
-
---log-level
-optional
+```properties
+# vmargs=-Xmx1g -XX:MaxDirectMemorySize=512m -Dlog4j.configuration=/opt/ml/conf/log4j.properties
```
Comment (frankfliu) [inline docker/README.md]: file:///

File: docker/advanced_settings.md [hunk 10]
```diff
@@ -335,111 +294,43 @@ The predict endpoint will return a prediction response in JSON. It will look som
...
+* **keystore** - SSL Key Store
+* **keystore_pass** - SSL password
+* **keystore_type** - Store of cryptographic keys and certificates
+* **private_key_file** - Location of the private key file
+* **certificate_file** - Location of the certificate file
+
+in the range of 0 .. (num-gpu-1) in a round-robin fashion. **By default MMS uses all the available GPUs but this parameter can be configured if user want to use only few of them**.
+
+```properties
+vmargs=-Xmx1g -XX:MaxDirectMemorySize=512m -Dlog4j.configuration=/opt/ml/conf/log4j.properties
```
Comment (frankfliu) [inline docker/advanced_settings.md]: file:///

File: docker/advanced_settings.md (hunk 2 above)
Comment (frankfliu) [reply docker/advanced_settings.md]: We don't want to misleading user even in example.
keystore related parameter won't be used for http
and it doesn't make sense both key file and keystore ad defined

File: docker/advanced_settings.md [hunk 11]
```diff
@@ -129,29 +110,39 @@ Interact with the container. This will open a shell prompt inside the container.
...
-Start MMS in the Docker container (GPU config):
+Start MMS in the Docker container using nvidia-docker command as follows. :
 ```bash
-nvidia-docker exec mms bash -c "mxnet-model-server start --mms-config /models/mms_app_cpu.conf"
+nvidia-docker exec mms bash -c mxnet-model-server --start --mms-config /home/model-server/config.properties
+```
+
+**Note**: To use GPU configuration, modify the config.properties to reflect that the model-server should use GPUs.
+
+```yaml
```
Comment (frankfliu) [inline docker/advanced_settings.md]: We should use ```properties

File: docker/README.md (hunk 9 above)
Comment (vdantu) [reply docker/README.md]: True, will make this change.

File: docker/advanced_settings.md (hunk 11 above)
Comment (vdantu) [reply docker/advanced_settings.md]: I think I already made this change

Comment (vdantu) [reply docker/advanced_settings.md]: Oh, I think I missed one. Made the change now.

File: docker/README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (vdantu) [reply docker/README.md]: Sure, lets take it up as a second pass over the doc. This change would touch multiple things in the doc.

**Pull request merged**